scripts/format_notebooks.py

```diff
@@ -181,6 +181,58 @@ def preprocess(self, nb, resources):
         return nb, resources
 
 
+class VersionCheckPreprocessor(preprocessors.Preprocessor):
+    metadata_tag = "VersionCheck"  # tag for added cells so that we can find them easily
+
+    try:
+        # determine the current stellargraph version by importing the library's version file
+        sys.path.append(
+            (
+                os.path.abspath(
+                    os.path.join(
+                        os.path.dirname(__file__), os.path.pardir, "stellargraph"
+                    )
+                )
+            )
+        )
+        import version as sgversion
+
+        stellargraph_version = sgversion.__version__
+    except ImportError as error:
+        print(f"ERROR: unable to import StellarGraph version: {error.message}")
+        stellargraph_version = None
+
+    version_check_code = f"""\
+# verify that we're using the correct version of StellarGraph for this notebook
+import stellargraph as sg
+from packaging import version
+
+if version.parse(sg.__version__) < version.parse("{stellargraph_version}"):
+    raise ValueError(
+        f"The installed version {{sg.__version__}} of StellarGraph is older than the version {stellargraph_version} required for this notebook. Please update StellarGraph, eg: pip install --upgrade stellargraph"
+    )"""
+
+    def preprocess(self, nb, resources):
+        # remove any cells we added in a previous run
+        nb.cells = [
+            cell
+            for cell in nb.cells
+            if self.metadata_tag not in cell["metadata"].get("tags", [])
+        ]
+        # find first (non-CloudRunner) code cell and insert before it
+        first_code_cell_id = next(
+            index
+            for index, cell in enumerate(nb.cells)
+            if cell.cell_type == "code"
+            and CloudRunnerPreprocessor.metadata_tag
```

Comment:
We could factor out this tag checking logic, potentially?

E.g.

```python
class CloudRunnerPreprocessor(...):
    @staticmethod
    def has_tag(cell):
        return CloudRunnerPreprocessor.metadata_tag not in cell["metadata"].get("tags, [])
```

Or even just a helper function at the top level that can be used everywhere

```python
def tags(cell):
    return cell["metadata"].get("tags", [])

...

# in both CloudRunnerPreprocessor and VersionCheckPreprocessor
nb.cells = [
    cell for cell in nb.cells if self.metadata_tag not in tags(cell)
]
```

Reply:
This is a good idea, and I think even more can be shared via a common class